src/hatti/views/map.cljs

```diff
@@ -489,6 +475,85 @@
                    :on-click (click-fn #(om/update! geofield field))}]
                  (get-label field) [:br]])]]]))))))
 
+(defn map-hexbin-selector
+  [{{{:keys [show?]} :hexbins} :map-page :as cursor} owner]
+  (reify
+    om/IRenderState
+    (render-state [_ _]
+      "Render hexbin selector component w/css + expansion technique from
+      leaflet layer control."
+      (html
+       [:div.leaflet-left.leaflet-bottom.hexbin-selector
+        [:div.leaflet-control.leaflet-control-layers
+         [:a
+          {:title "Hexbins Layer"
+           :class (str "hexbin-layer-toggle" (when show? " active"))
+           :on-click
+           (click-fn
+            #(om/update! cursor
+                         [:map-page :hexbins :show?] (not show?)))}]]]))))
+
+(defn map-hexbin-slider
+  [{{{:keys [show? cell-width hide-points?]} :hexbins} :map-page :as cursor}
+   owner]
+  (reify
+    om/IInitState
+    (init-state [_]
+      {:slider-value 0
+       :zoom->bin? true
+       :value->size {0 500 1 250 2 100 3 50 4 25 5 10 6 5 7 1 8 0.5}})
+    om/IWillReceiveProps
+    (will-receive-props [_ next-props]
+      (when (om/get-state owner :zoom->bin?)
+        (when-let [map (-> next-props :map-page :mapboxgl-map)]
+          (let [zoom (.getZoom map)
+                value (cond
```

Comment:
can be written as

```clojure
(condp #(<= %2 %1) zoom
  2 0
  5 1
  8 2
  11 3
  14 4
  17 5
  20 6)
```

but that will raise an exception for `> 20` and return 0 for `< 0`.

```clojure
(condp #(<= %2 %1) zoom
  -1 nil
  2  0
  5  1
  8  2
  11 3
  14 4
  17 5
  20 6
  nil)
```

will exactly duplicate the existing functionality